riscv/src/runtime.rs

```diff
@@ -73,635 +67,132 @@ impl SubMachine {
-        );
-
-        r.add_submachine::<&str, _, _>(
-            "std::machines::range::Bit2",
-            None,
-            "bit2",
-            vec![],
-            [],
-            0,
-            [],
-        );
-
-        r.add_submachine::<&str, _, _>(
-            "std::machines::range::Bit6",
-            None,
-            "bit6",
-            vec![],
-            [],
-            0,
-            [],
-        );
-
-        r.add_submachine::<&str, _, _>(
-            "std::machines::range::Bit7",
-            None,
-            "bit7",
-            vec![],
-            [],
-            0,
-            [],
-        );
-
-        r.add_submachine::<&str, _, _>(
-            "std::machines::range::Byte",
-            None,
-            "byte",
-            vec![],
-            [],
-            0,
-            [],
-        );
-
-        r.add_submachine::<&str, _, _>(
-            "std::machines::range::Byte2",
-            None,
-            "byte2",
-            vec![],
-            [],
-            0,
-            [],
-        );
-
-        r.add_submachine::<&str, _, _>(
-            "std::machines::binary::ByteBinary",
-            None,
-            "byte_binary",
-            vec![],
-            [],
-            0,
-            [],
-        );
-
-        r.add_submachine::<&str, _, _>(
-            "std::machines::shift::ByteShift",
-            None,
-            "byte_shift",
-            vec![],
-            [],
-            0,
-            [],
-        );
-
-        r.add_submachine::<&str, _, _>(
-            "std::machines::split::ByteCompare",
-            None,
-            "byte_compare",
-            vec![],
-            [],
-            0,
-            [],
-        );
-
-        // Base syscalls
-        r.add_syscall(
-            // TODO this is a quite inefficient way of getting prover inputs.
-            // We need to be able to access the register memory within PIL functions.
-            Syscall::Input,
-            [
-                "query_arg_1 <== get_reg(10);",
-                "query_arg_2 <== get_reg(11);",
-                "set_reg 10, ${ std::prelude::Query::Input(std::convert::int(std::prover::eval(query_arg_1)), std::convert::int(std::prover::eval(query_arg_2))) };",
-            ]
-        );
-
-        r.add_syscall(
-            Syscall::Output,
-            // This is using x0 on purpose, because we do not want to introduce
-            // nondeterminism with this.
-            [
-                "query_arg_1 <== get_reg(10);",
-                "query_arg_2 <== get_reg(11);",
-                "set_reg 0, ${ std::prelude::Query::Output(std::convert::int(std::prover::eval(query_arg_1)), std::prover::eval(query_arg_2)) };"
-            ]
-        );
+pub struct SyscallImpl(pub Vec<FunctionStatement>);
 
-        r.add_syscall(Syscall::Halt, ["return;"]);
+use crate::runtime_16::Runtime16;
+use crate::runtime_32::Runtime32;
 
-        r
-    }
+pub enum RuntimeEnum {
```

Comment:
idea: how about a `RuntimeOptions` type, which must be passed to the `compile/translate` functions? It could contain all information needed to compile and instantiate a runtime. It could have `KnownField`, `with_bootloader`, whatever coprocessors are needed, and then, only inside when the runtime is actually needed, call a `build32|16` to create the concrete runtime

Reply:
inside the compilation method you have all info needed to instantiate the correct runtime (e.g., does it need `poseidon_for_continuations`?)

Reply:
Will do!